pkg/karmadactl/cmdinit/utils/util.go

```diff
@@ -8,9 +8,38 @@ import (
 	"net/http"
 	"os"
 	"path/filepath"
+	"strings"
 	"time"
 )
 
+// InitVersion init karmada version
+type InitVersion struct {
+	ImageVersion string
+	CRDVersion   string
+}
+
+// GetInitVersion formatted version
+func GetInitVersion(version string) InitVersion {
+	// CRD version e.g. v0.10.1-547-g0190fda9-dirty -> v0.10.0
+	n := strings.LastIndex(version, ".") + 1
+
+	return InitVersion{
+		ImageVersion: initImageVersion(version),
+		CRDVersion:   fmt.Sprintf("%s%s", string([]byte(version)[:n]), "0"),
+	}
+}
```

Comment:
I guess there will be some tools/packages to handle the git version thing. I'll try to investigate it.